packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,361 @@
+      isAmbiguousTime: false,
+      isTimeClarificationOpen: false,
+    };
+
+    // The dateValue and timeValue variables represent the actual value in the date input and time input respectively.
+    // They are used to keep track of the currently entered value to determine whether or not the entry is valid.
+    // Unlike dateValue and timeValue, this.state.dateTime is the internal moment object representing both the date and time as one entity
+    // It is used for date/time minuipulation and used to calculate the missing/ambiguous hour.
+    // The dateValue and timeValue are tracked outside of the react state to limit the number of renderings that occur.
+    this.dateValue = '';
+    this.timeValue = '';
+
+    this.handleDateChange = this.handleDateChange.bind(this);
+    this.handleDateChangeRaw = this.handleDateChangeRaw.bind(this);
+    this.handleTimeChange = this.handleTimeChange.bind(this);
+    this.handleOnSelect = this.handleOnSelect.bind(this);
+    this.handleOnDateBlur = this.handleOnDateBlur.bind(this);
+    this.handleOnTimeBlur = this.handleOnTimeBlur.bind(this);
+
+    this.handleDaylightSavingButtonClick = this.handleDaylightSavingButtonClick.bind(this);
+    this.handleStandardTimeButtonClick = this.handleStandardTimeButtonClick.bind(this);
+  }
+
+  componentWillMount() {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+
+    this.dateValue = DateTimeUtils.formatMomentDateTime(this.state.dateTime, dateFormat);
+    this.timeValue = DateTimeUtils.hasTime(this.props.value) ? DateTimeUtils.formatISODateTime(this.props.value, 'HH:mm') : '';
+  }
+
+  handleOnSelect(event, selectedDate) {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    this.dateValue = DateTimeUtils.formatISODateTime(selectedDate, dateFormat);
+    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    const updatedDateTime = DateTimeUtils.syncDateTime(previousDateTime, selectedDate, this.timeValue);
+
+    if (!previousDateTime || previousDateTime.format() !== updatedDateTime.format()) {
+      this.checkAmbiguousTime(updatedDateTime);
+    }
+  }
+
+  handleOnDateBlur(event) {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    const isDateTimeValid = DateTimeUtils.isValidDateTime(event.target.value, this.timeValue, dateFormat);
+    const enteredDateTime = isDateTimeValid ? this.state.dateTime : null;
+
+    this.checkAmbiguousTime(enteredDateTime);
+  }
+
+  handleOnTimeBlur() {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    const isDateTimeValid = DateTimeUtils.isValidDateTime(this.dateValue, this.timeValue, dateFormat);
+    let updatedDateTime;
+
+    if (isDateTimeValid) {
+      updatedDateTime = DateTimeUtils.updateTime(this.state.dateTime, this.timeValue);
+    }
+
+    this.checkAmbiguousTime(updatedDateTime);
+  }
+
+  checkAmbiguousTime(dateTime) {
```

Comment:
I'm not quite convinced that this component should be handling the ambiguous time frame. Doing so means that you are making assumptions of time zone or basing it off the browser's time zone and I'm not really sure every consumer will want that behavior. For instance, HealtheIntent wants to handle all times in UTC.

At a minimum, I think this behavior should be opt-in for consumers or broken off to another wrapper component and consumers that want to deal with it can.

Reply:
I think that is a valid point. I'll bring this up in our war room discussion and see what others think about making this an opt-in behavior.

Reply:
The decision from the war room was this component will honor the ambiguous hour functionality by default and when there is a need to exclude this we can then add an opt-out option passively.